drgn_tools/kernfs_memcg.py

```diff
@@ -198,17 +213,28 @@ def dump_page_cache_pages_pinning_cgroups(
                 if max_pages and found_count == max_pages:
                     break
         except FaultError:
+            fault_count = fault_count + 1
             continue
 
     print(
-        f"Scanned {total_count} pages, found {found_count} pages with memory cgroup refs."
+        f"Scanned {total_count} pages, found {found_count} pages with memory cgroup refs, found {fault_count} faults."
     )
 
 
+class NumMemCgroups(CorelensModule):
+    """Print number of active and dying memcgroups"""
+
+    name = "num-memcgroups"
+
+    def run(self, prog: Program, args: argparse.Namespace) -> None:
+        get_num_mem_cgroups(prog)
+
```

Comment:
Can we make this more generic like the output of `cat /proc/cgroups`, not only support memory cgroup, but other cgroup as well? It doesn't have to be include with this pull request, you can start a new one for it if you want.

Reply:
Sure. I can include this change in a different PR. I included memcgroup numbers here because we often run into issue due to zombie memcgroups and hence having their number readily available will help us to decide if we need to run `PagesPinningMemcgroups` module.